src/components/CodeEditorWindow/config/RequesFromCode.js

```diff
@@ -0,0 +1,214 @@
+import axios from 'axios';
+
+function detectCURLinLine(line) {
+  // returns true if text matches a curl request
+  return line.match(/^\s*?curl\s+(-X[A-Z]+)?\s*['"]?.*?['"]?(\s*$|\s+?-d\s*?['"])/);
+}
+
+export function detectCURL(text) {
+  // returns true if text matches a curl request
+  if (!text) return false;
+  for (const line of text.split('\n')) {
+    if (detectCURLinLine(line)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+export  function  RequestFromCode(text) {
```

Comment:
This function is too complicated.
What I would suggest to do:

- split this function into multiple smaller ones
- for each function create a docstring comment with explanation what this function does + example of input and output
- Create tests, to the function could be tested on a variety on inputs without the need to open UI and manually copy-paste things


Additioanlly, it seems that you are trying to manually parse code with a huge number of regex. That's not the best approach in a long run, so I would suggest to try to find some ready-made libraries which could do it for you.